app/gui/src/dashboard/services/RemoteBackend.ts

```diff
@@ -1471,6 +1471,36 @@ export default class RemoteBackend extends Backend {
     }
   }
 
+  /** Download the project to a temporary location. */
+  async downloadProject(id: backend.ProjectId): Promise<DirectoryId> {
+    const details = await this.getProjectDetails(id, true)
+
+    invariant(details.url != null, 'The download URL of the project must be present.')
```

Comment:
prefer passing id directly using parameters instead.

Reply:
The `invariant` method only silences the compiler. `getPropertyDetails` should always return `url` property when called with the parameter `true`. It is used this way in the `download` method above, for example